packages/xo-server/src/xo-mixins/xen-servers.js

```diff
@@ -206,6 +207,21 @@ export default class {
     const objects = this._xo._objects
 
     forEach(newXapiObjects, function handleObject(xapiObject, xapiId) {
+      const serverIdsByPool = this._serverIdsByPool
+
+      // handle pool UUID change
+      if (
+        xapiObject.$type === 'pool' &&
+        serverIdsByPool[xapiObject.$id] === undefined
+      ) {
+        const obsoletePoolId = findKey(
+          serverIdsByPool,
+          serverId => serverId === conId
+        )
+        delete serverIdsByPool[obsoletePoolId]
+        serverIdsByPool[xapiObject.$id] = conId
+      }
```

Comment:
LGTM but we should find a way to test it.

Reply:
Maybe we could add some test code to call `Xapi#_setUrl()` like I did for my tests?

Reply:
So, it will be `xen-servers.spec.js` file?